crates/bevy_picking_core/src/focus.rs

```diff
@@ -0,0 +1,118 @@
+//! Determines which entities are being hovered by pointers, taking into account [`FocusPolicy`],
+//! [`RenderLayers`], and entity depth.
+
+use std::collections::BTreeMap;
+
+use crate::{backend, output::PointerInteraction, pointer::PointerId};
+use bevy::{
+    prelude::*,
+    render::view::{Layer, RenderLayers},
+    ui::FocusPolicy,
+    utils::{FloatOrd, HashMap, HashSet},
+};
+
+/// A map of entities sorted by depth.
+type DepthMap = BTreeMap<FloatOrd, Entity>;
+
+/// Maps [`RenderLayers`] to the map of entities within that pick layer, sorted by depth.
+type LayerMap = BTreeMap<Layer, DepthMap>;
+
+/// Maps Pointers to a [`LayerMap`]. Note this is much more complex than the [`HoverMap`] because
+/// this data structure is used to sort entities by layer then depth for every pointer.
+type OverMap = HashMap<PointerId, LayerMap>;
+
+/// Maps pointers to the entities they are hovering over. "Hovering" refers to the actual hover
+/// interaction state, in this case, not just whether the pointer happens to be over the entity.
```

Comment:
"Over" = the pointer passes a hit test, and is within the bounds of the object
"Hover" = the pointer is over the object AND the pointer has done a depth test to see if this object should be in its hover state. The depth test looks at the topmost object under the cursor, and checks iteratively to see if the pointer should "pass through" to the next entity, stopping when it reaches the first object it can interact with.

Ex: UI drawn over a viewport. The pointer is "over" a UI button and a mesh in the scene. However, only the UI button will be in the "hover" state.

Reply:
Yea that makes complete sense. I realized this after reading the systems that build the `HoverMap`. It might not be necessary to add this much detail to _this_ doc string, but maybe just a short explanation of what "hover interaction" actually means would be nice.